[jmx][loader] enable legacy JMX check loading

truthbk opened the pull request:
### What does this PR do?

This PR modifies the check loader to allow loading of JMX configurations similarly to the legacy behavior.

Bumps jmxFetch to the latest version, and includes the environment variable to allow building on gitlab.

It also strips some unnecessary logic, and removes the IPC pipe, as that will be replaced by the gRPC server/client architecture we'll be rolling out in the very near future.

### Motivation

In reality JMX is more a component than a check, and it has a N:1 configuration to check ratio. We could've removed the check from the catalog, but it wouldn't have changed much as the jmxLoader will still have to configure the long-running check. We can maybe clean this up a bit when time is less pressing. 

Some design decisions were made in agent5 that probably didn't make too much sense and we might be able to address those in the future, but for now, to enable a seamless upgrade, and for simplicity we should aim to mimic as sanely as possible the legacy behavior. This entails parsing the enabled jmx YAML configurations and do our best to launch the JMXFetch jar with the specified values. The JMXFetch check is implemented as a singleton - the first configuration loaded will actually instantiate the check, and subsequent configurations will allow further configuration of the one check. Finally, the long-running check will be launched, and JMXFetch will then load all specified configurations. 

In the future, configurations will be provided to JMXFetch via the gRPC interface, but we will still have to decide how to launch JMXFetch. Although I highly dislike the fact that these options might be disseminated over several JMX check config YAMLs, that the reality our customers live in. In my book these options should be top-level configuration options.

truthbk committed:
[jmx] launch legacy configs.

truthbk committed:
[lint] export CheckFactory.

truthbk committed:
[jmx] disabling test.

truthbk committed:
[jmx] implement singleton behavior.

truthbk committed:
[jmx][test] disabling.

truthbk committed:
[jmx] fix parser, recover in case of parser crash.

truthbk committed:
[jmx] skip jmx checks in go check loader.

truthbk committed:
[jmx] attempt to configure check as specifically as possible.

truthbk committed:
[jmx] running flag must be atomic.

truthbk committed:
[omnibus][gitlab] use default jmxfetch software definition.

truthbk committed:
[jmx] lint.

truthbk committed:
[jmx] pinning version to updated 0.16.0.

truthbk committed:
[jmxloader] kill pipe, ipc will be handled by future grpc server

truthbk committed:
[jmxloader] fix tests

truthbk committed:
[jmx][loader] fixing test.

truthbk reviewed (commented):
I didn't overthink it too much as this was urgent and I'll be continuing work on this same code in the next couple of days with the grpc stuff, but there could be some more cleanup in order.

olivielpeau reviewed (dismissed):
Made a few comments, but overall I think this is going to work fine, thanks for taking care of this! Feel free to merge once the comments are adressed (and the disabled test is fixed).

One general comment: any reason for keeping the jmx check in the corecheck catalog? Since it should never be loaded by the `GoCheckLoader` I'm not sure what made you keep the check in that catalog.

And just a general comment for the future GRPC interface: currently I don't think `AutoConfig` correctly handles the case where one Check struct has multiple configs, in particular when configs change and it tries to reload the correct checks.

truthbk committed:
[jmx] cleanup + custom omnibus branch.

truthbk committed:
[jmx] fix tests for jmx parser.

truthbk committed:
[jmx] move running atomic flag to the right location

truthbk committed:
[omnibus] restoring to master branch for omnibus-software

olivielpeau reviewed (dismissed):
Left one comment, I'm going to make the related change and merge

olivielpeau committed:
[jmx][omnibus] Define env vars in invoke task

So that the `agent.omnibus` invoke task doesn't require setting env
vars before being run.

olivielpeau committed:
Merge branch 'master' into jaime/jmxfix

olivielpeau committed:
[jmx] Remove env var definition from .gitlab-ci.yaml
